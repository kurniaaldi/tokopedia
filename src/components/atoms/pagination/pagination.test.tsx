import { render, screen } from "@testing-library/react";
import Pagination from "./index";
import renderer from "react-test-renderer";

test("pagination", () => {
  let data = {
    total: 5000,
    currentPage: 2,
    hasNextPage: true,
  };

  render(
    <Pagination
      total={data.total}
      next={(page: number) => console.log(page)}
      currentPage={data.currentPage}
      previous={(page: number) => console.log(page)}
      hasNextPage={data.hasNextPage}
    />
  );

  const dialogElement = screen.getByTestId("pagination-id");
  expect(dialogElement).toBeInTheDocument();
  expect(dialogElement).toHaveTextContent("2");
});

test("matches snapshot", () => {
  let data = {
    total: 5000,
    currentPage: 2,
    hasNextPage: true,
  };

  const tree = renderer
    .create(
      <Pagination
        total={data.total}
        next={(page: number) => console.log(page)}
        currentPage={data.currentPage}
        previous={(page: number) => console.log(page)}
        hasNextPage={data.hasNextPage}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
