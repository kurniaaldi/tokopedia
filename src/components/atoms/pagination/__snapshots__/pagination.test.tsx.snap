// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="pagination"
  data-testid="pagination-id"
>
  <li
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    «
  </li>
  <li
    className="active"
  >
    2
  </li>
  <li
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    »
  </li>
</div>
`;
